pkg/controller/cluster/cluster_controller.go

```diff
@@ -137,11 +138,10 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	klog.Infof("reconciling cluster object %v triggers idempotent reconcile.", name)
-	err = r.actuator.Reconcile(cluster)
-	if err != nil {
-		if requeueErr, ok := err.(*controllerError.RequeueAfterError); ok {
+	if err := r.actuator.Reconcile(cluster); err != nil {
+		if requeueErr, ok := errors.Cause(err).(controllerError.HasRequeueAfterError); ok {
```

Comment:
For others that might not know: from the docs of `errors.Cause` 
> If the error does not implement Cause, the original error will be returned. If the error is nil, nil will be returned without further investigation.

Reply:
Hi @vincepri,

Thank you. I actually had to go look up `Cause` to see what it did and found the same. The fact that it handles `nil` and even returns the original error made this very efficient to implement.

Reply:
*Technically* lines 141 and 142 (and the others like them) could be combined, but I think the way it is more readable.